fe1-web/components/TextInputList.tsx

```diff
@@ -0,0 +1,78 @@
+import React, { useState } from 'react';
+import PropTypes from 'prop-types';
+import { View } from 'react-native';
+import RemovableTextInput from './RemovableTextInput';
+
+/**
+ * Component which creates a list of Text input fields which can be deleted individually
+ * Returns array with non-empty Values of the text fields
+ * It uses the RemovableTextInput.tsx component to show the individual inputs
+ * Input: A function that stores the values that were inputted by the user
+ * Output: Array of the unique, nonempty values inputted by the user
+ */
+
+const TextInputList = (props: IPropTypes) => {
+  const { onChange } = props;
+  const [idCount, setIdCount] = useState(1);
```

Comment:
I think you could do without this, by simply referencing the length of userOptions, and using `map( (val,idx) => { ...} )` when you need to iterate.

On the other hand, if this is what you've tested and it works, don't waste extra time doing this for now :)

Reply:
I think we need a unique ID for each textField. 
When the `RemovableTextInput` gets created it gets the Id as a parameter from `TextInputList` which it uses in the removeInput and updateInput function
So if we delete an entry in the middle of the list of textfields, the indexes don't match anymore.  

**1. Current state**
Input_0, index: 0, remove(0)
Input_1, index: 1, remove(1)
Input_2, index: 2, remove(2)

**2. Delete Input_1**
Input_0, index: 0, remove(0)
Input_2, index: 1, remove(2)

**3. Try to delete Input_2**
remove(2) doesn't work since no Item there

At least this is what happens in my implementation. 
Perhaps by refactoring the logic of the functions it is possible to use only indexes, but I thought that having unique IDs would be better.

Reply:
Fair enough. Thanks for the rationale 👍